go/interpreter/lfvm/opcode.go

```diff
@@ -497,3 +352,25 @@ func (o OpCode) decompose() []OpCode {
 	}
 	return nil
 }
+
+// IsValid returns true if the OpCode is a valid OpCode.
+// A valid opcode shall be defined in the OpCode space, with the exception of
+// INVALID, which is a defined value.
+func (op OpCode) isValid() bool {
+	if op < FIRST_LFVM_EXTENDED_OPCODE {
+		return vm.IsValid(vm.OpCode(op))
+	}
+	return op >= FIRST_LFVM_EXTENDED_OPCODE && op < END_LFVM_EXTENDED_OPCODES
+}
+
+// isExecutable returns true if the OpCode is an executable OpCode.
+// An executable OpCode is a valid OpCode that executes an operation; all valid
+// OpCodes with the exception of DATA, and NOOP are executable.
+// - JUMPDEST is an executable OpCode.
+// - INVALID is not an executable OpCode, since it is neither Valid
+func (op OpCode) isExecutable() bool {
+	if op < FIRST_LFVM_EXTENDED_OPCODE {
+		return op.isValid()
+	}
+	return op < END_LFVM_EXECUTABLE_OPCODES
+}
```

Comment:
I think these functions can be eliminated by defining the following:
- OpCodes are 9-bit values; the first 7 bits are to be ignored, only the trailing 9 bits are relevant
- every possible OpCode is a "valid" opcode (the full range of 0-512)
- what is "executable" is of nobodies business but the interpreter loop
- we provide constants for the 7-bit `opCodeMask` and the derived `numOpCodes` for convenience to implement property maps like the static gas or the stack usage map

I think this further simplifies the definition. In particular, by dropping the distinction between valid and invalid op-codes, some functions also get rid of error cases. I prepared a PR targeting this branch show-casing how this is playing out. Let me know what you think.

Reply:
Unfortunately not all opcodes in rage 0-512 are valid.  there are gaps in the definition with values that are neither an op-code nor INVALID opcode.

Reply:
Function was removed.